pkg/tsdb/phlare/query.go

```diff
@@ -369,24 +368,68 @@ func treeToNestedSetDataFrame(tree *ProfileTree, profileTypeID string) *data.Fra
 	parts := strings.Split(profileTypeID, ":")
 	valueField.Config = &data.FieldConfig{Unit: normalizeUnit(parts[2])}
 	selfField.Config = &data.FieldConfig{Unit: normalizeUnit(parts[2])}
-	labelField := data.NewField("label", nil, []string{})
 	lineNumberField := data.NewField("line", nil, []int64{})
-	fileNameField := data.NewField("fileName", nil, []string{})
-	frame.Fields = data.Fields{levelField, valueField, selfField, labelField, lineNumberField, fileNameField}
+	frame.Fields = data.Fields{levelField, valueField, selfField, lineNumberField}
+
+	labelField := NewEnumField("label", nil)
+	fileNameField := NewEnumField("fileName", nil)
 
 	walkTree(tree, func(tree *ProfileTree) {
+
 		levelField.Append(int64(tree.Level))
 		valueField.Append(tree.Value)
 		selfField.Append(tree.Self)
 		// todo: inline functions
 		// tree.Inlined
-		labelField.Append(tree.Function.FunctionName)
 		lineNumberField.Append(tree.Function.Line)
+		labelField.Append(tree.Function.FunctionName)
 		fileNameField.Append(tree.Function.FileName)
 	})
+
+	frame.Fields = append(frame.Fields, labelField.GetField(), fileNameField.GetField())
 	return frame
 }
 
+type EnumField struct {
```

Comment:
@leeoniya I assume something like this would make sense in the backend sdk? I am just not keen of this as an API so will probably think about it a bit, or if you have some suggestions let me know.

Reply:
tagging @ryantxu in for this one

Reply:
Yes -- something like this should be added to the backend SDK.  The SDK currently supports the transport side of things, but does not have helpers to make it easy to use.  

Perhaps worth exploring ergonomic APIs here, then moving to the SDK?